### PR TITLE
Feature: Add Lambda, ECS, AppRunner, and Lightsail examples

### PR DESCRIPTION
**Pull request by mattmoor**
:gift: This adds an example showing how to use tf-ko with AWS Lambda, ECS, AppRunner, and Lightsail.

/kind feature

### CODE REVIEW

File: provider-examples/ecs/ecs.tf [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  network_mode             = "awsvpc"
+  execution_role_arn       = aws_iam_role.example.arn
+  cpu                      = 1024
+  memory                   = 2048
+  container_definitions    = jsonencode([
+    {
+      "name": "foo",
+      # DO NOT SUBMIT
+      "image": aws_ecr_repository.foo.repository_url,
+      # "image": ko_image.image.image_ref,
```
Comment (mattmoor) [inline provider-examples/ecs/ecs.tf:105]: @imjasonh this is my workaround for now.  If I manually push the image, things work with this, so we should be *really* close!

Comment (imjasonh) [reply provider-examples/ecs/ecs.tf:105]: This should be good to go now with #48 in v0.0.6 🤞

File: provider-examples/ecs/ecs.tf [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    scan_on_push = false
+  }
+}
+
+resource "ko_image" "image" {
+  repo        = aws_ecr_repository.foo.repository_url
+  base_image  = "cgr.dev/chainguard/static:latest-glibc"
+  importpath  = "github.com/ko-build/terraform-provider-ko/cmd/test"
+  working_dir = path.module
+  // ko's SBOMs are being rejected by ECR currently, so disable SBOMs.
```
Comment (imjasonh) [inline provider-examples/ecs/ecs.tf:40]:
```suggestion
  // Disable SBOM generation due to
  // https://github.com/ko-build/ko/issues/878
```

File: provider-examples/ecs/ecs.tf (hunk 1 above)
Comment (mattmoor) [reply provider-examples/ecs/ecs.tf:105]: Will give it a try just as soon as TF finds the new version :-/

File: provider-examples/ecs/ecs.tf (hunk 2 above)
Comment (mattmoor) [reply provider-examples/ecs/ecs.tf]: done

File: cmd/test-lambda/main.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import "github.com/aws/aws-lambda-go/lambda"
+
+func hello() (string, error) {
+	return "Hello ƛ!", nil
+}
+
+func main() {
+	// Make the handler available for Remote Procedure Call by AWS Lambda
+	lambda.Start(hello)
```
Comment (mattmoor) [inline cmd/test-lambda/main.go:11]: Shamelessly stolen from @imjasonh here: https://github.com/imjasonh/lambda-ko.

Thanks for the pointer!

**Review by imjasonh — APPROVED** [review]

**Comment by mattmoor** [comment]
Ok, it finally found 0.0.6.  I'm testing each of these now

**Comment by mattmoor** [comment]
These all work, I'm removing the WIP and enabling auto-merge

**Pull request merged**